src/Compilers/Core/Portable/Emit/EditAndContinue/SymbolChanges.cs

```diff
@@ -284,7 +314,26 @@ private static void CalculateChanges(IEnumerable<SemanticEdit> edits, out IReadO
                         break;
 
                     case SemanticEditKind.Delete:
-                        // No work to do.
+                        // We allow method deletions only at the moment
+                        if (edit.OldSymbol is IMethodSymbol && edit.NewContainingSymbol is not null)
+                        {
+                            Debug.Assert(edit.OldSymbol != null);
+                            lazyDeletedSymbolsBuilder ??= new();
+                            if (!lazyDeletedSymbolsBuilder.TryGetValue(edit.NewContainingSymbol, out var set))
+                            {
+                                set = new HashSet<ISymbol>();
+                                lazyDeletedSymbolsBuilder.Add(edit.NewContainingSymbol, set);
+                            }
+                            set.Add(edit.OldSymbol);
+                            // If a type has a single change that is a deletion, then nothing will be emitted
+                            // for it, so we need to make sure we track the containing type of the delection
```

Comment:
"If a type has a single change that is a deletion, then nothing will be emitted for it" describes the previous state of the code that's not valid anymore once you make the change.